cmd/syncEnumerator.go

```diff
@@ -211,6 +213,14 @@ func (cca *cookedSyncCmdArgs) initEnumerator(ctx context.Context) (enumerator *s
 	}
 }
 
+func IsLocationCaseInsensitive(fromTo common.FromTo) bool {
+	if fromTo.IsDownload() && runtime.GOOS == "windows" {
```

Comment:
I tried it on Mac and it has the same problem, have you tried on Linux just to be safe?

Reply:
It shouldn't work with UNIX based systems because, in UNIX, path is case-sensitive.
So a user may have two distinct paths -
`/home/mohsha/go_projects/src/azure-storage-azcopy`
`/home/mohsha/go_projects/SRC/Azure-Storage-Azcopy`

Notice they have same name but different casing and it is still allowed to create such paths.

With our approach, we lowercase the relative path and store it in the map. So we may end up thinking two or more different paths are actually the same.

Try that on Windows.

Reply:
Apple File System can be configured to be case sensitive or insensitive.
https://support.apple.com/en-in/guide/disk-utility/dsku19ed921c/mac

Reply:
We may have to deduce if the APFS is case-insensitive and then apply this logic.

Reply:
For a really dumb simple test, we could create two test files in the azcopy folder. This would be OS agnostic.